crypto/src/random/mod.rs

```diff
@@ -43,20 +50,21 @@ use utils::collections::Vec;
 /// assert_ne!(e2, e3);
 ///
 /// // should draw same elements for the same seed
-/// let mut coin1 = RandomCoin::<BaseElement, Blake3_256<BaseElement>>::new(&[1, 2, 3, 4]);
-/// let mut coin2 = RandomCoin::<BaseElement, Blake3_256<BaseElement>>::new(&[1, 2, 3, 4]);
+/// let mut coin2 = RandomCoinImpl::<BaseElement, Blake3_256<BaseElement>>::new(seed);
+/// let mut coin1 = RandomCoinImpl::<BaseElement, Blake3_256<BaseElement>>::new(seed);
 /// let e1 = coin1.draw::<BaseElement>().unwrap();;
 /// let e2 = coin2.draw::<BaseElement>().unwrap();;
 /// assert_eq!(e1, e2);
 ///
 /// // should draw different elements based on seed
-/// let mut coin1 = RandomCoin::<BaseElement, Blake3_256<BaseElement>>::new(&[1, 2, 3, 4]);
-/// let mut coin2 = RandomCoin::<BaseElement, Blake3_256<BaseElement>>::new(&[2, 3, 4, 5]);
+/// let mut coin1 = RandomCoinImpl::<BaseElement, Blake3_256<BaseElement>>::new(seed);
+/// let seed = Blake3_256::<BaseElement>::hash(&[2, 3, 4, 5]);
+/// let mut coin2 = RandomCoinImpl::<BaseElement, Blake3_256<BaseElement>>::new(seed);
 /// let e1 = coin1.draw::<BaseElement>().unwrap();;
 /// let e2 = coin2.draw::<BaseElement>().unwrap();;
 /// assert_ne!(e1, e2);
 /// ```
-pub struct RandomCoin<B, H>
+pub struct RandomCoinImpl<B, H>
```

Comment:
I would probably move `RandomCoin` trait here and the implementation into a sub-module. Also, maybe we should use something like `DefaultRandomCoin` instead of `RandomCoinImpl` to indicate that it just a default implementation?